### ISSUE TITLE
FR: Add filter field to the code blocks

### ISSUE THREAD

**Issue by HEmile** [SkepticMystic/breadcrumbs#262]
The code blocks are great! One caveat though is that they can get real big. And sometimes I want to query very specific things! Is it possible to add a filter field? 

One option could be to just use a Dataview query here, if you don't feel like implementing any annoying filtering options. Not sure how hard that would be. Maybe you can run both queries, and then compute their intersection.

**Comment by SkepticMystic** [comment]
I can definitely do that! That's what the `fields` field is for, I just haven't implemented it yet.
Btw, your issue you mentioned on the Discord is caused by this #265. Before I fix it, you can just add `fields: <field>` where `<field>` is _any_ field in your hierarchies. It will just bypass the error

**Comment by HEmile** [comment]
To be sure, this is a different request to what I asked on Discord. Ie: I don't mean selecting specific hierarchies (what I think `fields` is for, but more fine grained filter on eg tags, title, path... etc. The things you'd query in Dataview.

**Comment by SkepticMystic** [comment]
@HEmile would this cover your FR: #266

**Comment by SkepticMystic** [comment]
So `fields` would allow you to filter specific fields in the chosen `dir`. 
`where` will allow you to interact with the metadata on each result

**Comment by HEmile** [comment]
Yes, pretty much. The suggestion here is that you use Dataview queries inside the `where` field so that you don't have to implement a new filtering syntax.

**Comment by SkepticMystic** [comment]
How do you mean? Can I feed a query into it's API?

**Comment by SkepticMystic** [comment]
It seems `api.pages()` can query for tags or folders, but not for fields
https://blacksmithgu.github.io/obsidian-dataview/api/code-reference/

**Comment by HEmile** [comment]
Hmm, I expected DV to have implemented this by now, but you're right, it can only find Sources but not WHERE queries https://blacksmithgu.github.io/obsidian-dataview/query/sources/

**Comment by SkepticMystic** [comment]
Yeah... I don't think it'll be too hard to parse... I say that now

**Comment by SkepticMystic** [comment]
There is an FR for a `where` field #266, and a `from` field has been implemented, so I'm gonna close this out
